crates/api/api_utils/src/utils.rs

```diff
@@ -304,7 +304,7 @@ pub fn check_comment_deleted_or_removed(comment: &Comment) -> LemmyResult<()> {
 }
 
 pub async fn check_local_vote_mode(
-  score: i16,
+  is_upvote: Option<bool>,
```

Comment:
Its weird to me that the `is_upvote` column is a required bool, but in a lot of function signature's you've made it optional. IMO it'd be much clearer and less error prone if you made all these `bool`. 

Do a grep in this PR for `Option<bool>`, and change them to bool to match the DB.